js/clinicalTimeline.js

```diff
@@ -782,27 +786,23 @@ window.clinicalTimeline = (function(){
       var dayFormat = [];
       var m;
       var d;
-      if (time.y !== 0) {
-        dayFormat = dayFormat.concat(time.y+"y");
-      }
-      if (time.m !== 0) {
-        if (time.y !== 0) {
-          m = Math.abs(time.m);
-        } else {
-          m = time.m;
-        }
-        dayFormat = dayFormat.concat(m+"m");
-      }
-      if (time.d !== 0) {
-        if (time.y !== 0 || time.m !== 0) {
-          d = Math.abs(time.d);
-        } else {
-          d = time.d;
-        }
-        dayFormat = dayFormat.concat(d+"d");
-      }
       if (time.y === 0 && time.m === 0 && time.d === 0) {
         dayFormat = [0];
+      } else {
+        if (getZoomLevel(beginning, ending, width * zoomFactor) === "days" || getZoomLevel(beginning, ending, width * zoomFactor) === "10days" || getZoomLevel(beginning,ending,width) === "3days") {
+          d = time.toDays();
+          dayFormat = dayFormat.concat(d + "d");
+        } else if (getZoomLevel(beginning, ending, width * zoomFactor) === "months") {
+          m = time.m + 12 * time.y;
+          dayFormat = dayFormat.concat(m + "m");
+        } else if (getZoomLevel(beginning, ending, width * zoomFactor) === "years") {
+          y = time.y;
+          dayFormat = dayFormat.concat(y + "y");
+        } else {
+          //defaults to days as that is the minimum possible frame
+          d = time.toDays();
+          dayFormat = dayFormat.concat(d + "d");
+        }
```

Comment:
Great, this whole code block looks a lot cleaner now. I think you should change the concat just to an assignment e.g.: `dayFormat = d + "d";` because we are not growing the `dayFormat` anymore, we are just assigning the correct format at once. I also prefer the error you were logging before over defaulting to days. It'll be a little easier to detect if we accidentally generate an incorrect zoomLevel in the future